packages/dataviews/src/components/dataform/stories/index.story.tsx

```diff
@@ -366,7 +367,8 @@ function CustomEditControl< Item >( {
 
 	return (
 		<ValidatedTextControl
-			required={ !! field.isValid?.required }
+			required={ !! field.isValid.required }
+			customValidity={ validity?.custom ? validity.custom : undefined }
```

Comment:
What happens when a field has both a "required" check and an additional custom validation function?

Reply:
If the component gets a valid `customValidity` object, it'll be used to display the message. This can be used to override the native "required" message (and I've updated to make it so if one is provided).

However, the `required` prop has an additional function and cannot be removed. It adds a "(REQUIRED)" suffix to the field label, see:

<img width="536" height="98" alt="Screenshot 2025-10-15 at 11 45 39" src="https://github.com/user-attachments/assets/ddde9fbb-f39a-435d-ba5d-811ea800960a" />